http/vibe/http/fileserver.d

```diff
@@ -427,3 +418,214 @@ private void sendFileImpl(scope HTTPServerRequest req, scope HTTPServerResponse
 		logTrace("sent file %d, %s!", fil.size, res.headers["Content-Type"]);
 	}
 }
+
+/**
+	Calls $(D handleCache) with prefilled etag and lastModified value based on a file.
+
+	See_Also: handleCache
+
+	Returns: $(D true) if the cache was already handled and no further response must be sent or $(D false) if a response must be sent.
+*/
+bool handleCacheFile(scope HTTPServerRequest req, scope HTTPServerResponse res,
+		string file, string cache_control = null, Duration max_age = Duration.zero)
+{
+	return handleCacheFile(req, res, NativePath(file), cache_control, max_age);
+}
+
+/// ditto
+bool handleCacheFile(scope HTTPServerRequest req, scope HTTPServerResponse res,
+		NativePath file, string cache_control = null, Duration max_age = Duration.zero)
+{
+	if (!existsFile(file)) {
+		return false;
+	}
+
+	FileInfo ent;
+	try {
+		ent = getFileInfo(file);
+	} catch (Exception) {
+		throw new HTTPStatusException(HTTPStatus.internalServerError,
+				"Failed to get information for the file due to a file system error.");
+	}
+
+	return handleCacheFile(req, res, ent, cache_control, max_age);
+}
+
+/// ditto
+bool handleCacheFile(scope HTTPServerRequest req, scope HTTPServerResponse res,
+		FileInfo dirent, string cache_control = null, Duration max_age = Duration.zero)
+{
+	import std.bitmanip : nativeToLittleEndian;
+	import std.digest.md : MD5, toHexString;
+
+	SysTime lastModified = dirent.timeModified;
+	ETag etag;
+	etag.weak = dirent.isDirectory;
+	MD5 md5;
+	md5.put(lastModified.stdTime.nativeToLittleEndian);
+	md5.put(dirent.size.nativeToLittleEndian);
+	etag.tag = toHexString(md5.finish())[].idup;
```

Comment:
This block looks like it should be better wrapped into an `ETag` constructor or builder function?

Reply:
I just picked md5 and lastModified + dirent size as arbitrary choice (and because the previous algorithm was similar) and I think it would be better if this was transparent to the user so it could just be changed any time without breaking any code depending on the value of the algorithm used to generate the ETag

Changing the ETag algorithm (like the PR does) means that caches will become invalid and clients will refetch the version after the server updating vibe.d

So because the algorithm is not meant for generating a value to be used in user code and it just works with FileInfo anyway, I think it's ok to just keep it in the handleCacheFile function.

Reply:
Hmm, my point was more that this could be sth. like:

```d
struct ETag {
  static  ETag fromFile(FileInfo dirent) {
     ... (code from here)
     return etag;
  }
}
```

as imho it's not the job of this function to decide which hashing method to use.

Reply:
I added a helper function which does the hashing & string serialization now